persistence/pre_persistence/app.go

```diff
@@ -0,0 +1,311 @@
+package pre_persistence
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/pokt-network/pocket/shared/types"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"google.golang.org/protobuf/proto"
+)
+
+func (m *PrePersistenceContext) GetAppExists(address []byte) (exists bool, err error) {
+	db := m.Store()
+	key := append(AppPrefixKey, address...)
+	if found := db.Contains(key); !found {
+		return false, nil
+	}
+	bz, err := db.Get(key)
+	if err != nil {
+		return false, err
+	}
+	if bz == nil {
+		return false, nil
+	}
+	if bytes.Contains(bz, DeletedPrefixKey) {
+		return false, nil
+	}
+	return true, nil
+}
+
+func (m *PrePersistenceContext) GetApp(address []byte) (app *App, exists bool, err error) {
+	app = &App{}
+	db := m.Store()
+	key := append(AppPrefixKey, address...)
+	bz, err := db.Get(key)
+	if err != nil {
+		return nil, false, err
+	}
+	if bz == nil {
+		return nil, false, nil
+	}
+	if bytes.Contains(bz, DeletedPrefixKey) {
+		return nil, false, nil
+	}
+	if err = proto.Unmarshal(bz, app); err != nil {
+		return nil, true, err
+	}
+	return app, true, nil
+}
+
+func (m *PrePersistenceContext) GetAllApps(height int64) (apps []*App, err error) {
+	cdc := Cdc()
+	apps = make([]*App, 0)
+	var it iterator.Iterator
+	if height == m.Height {
```

Comment:
TODO: making a request to discuss this offline. Don't fully understand the design/scope.

Reply:
Let's do it